third_party/move/move-compiler/src/parser/ast.rs

```diff
@@ -509,15 +509,33 @@ pub enum Bind_ {
     Var(Var),
     // T { f1: b1, ... fn: bn }
     // T<t1, ... , tn> { f1: b1, ... fn: bn }
-    Unpack(Box<NameAccessChain>, Option<Vec<Type>>, Vec<(Field, Bind)>),
+    Unpack(
+        Box<NameAccessChain>,
+        Option<Vec<Type>>,
+        Vec<BindFieldOrDotdot>,
+    ),
     // T(e1, ..., en)
     // T<t1, ... , tn>(e1, ..., en)
-    PositionalUnpack(Box<NameAccessChain>, Option<Vec<Type>>, Vec<Bind>),
+    PositionalUnpack(Box<NameAccessChain>, Option<Vec<Type>>, Vec<BindOrDotdot>),
 }
 pub type Bind = Spanned<Bind_>;
 // b1, ..., bn
 pub type BindList = Spanned<Vec<Bind>>;
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum BindFieldOrDotdot_ {
+    FieldBind(Field, Bind),
```

Comment:
We cannot do that for `Bind` since we need to keep both `BindOrDotDot` and `Bind` which are needed in different contexts. Then for consistency, we would as well just keep `BindFieldOrDotDot` which save us also from changing
```rust
// Parse a field name optionally followed by a colon and a binding:
//      BindField = <Field> <":" <Bind>>?
...
fn parse_bind_field(context: &mut Context) -> Result<(Field, Bind), Box<Diagnostic>> {
```
in `syntax.rs`.

Reply:
FWIW, I think `XorY` is ok, though keeping it readable is an issue for `BindOrDotdot`.

Perhaps thinking of the meaning of `DotDot` rather than just the text: `..` represents an elided/omitted field name or names.  `OptionalFieldBinding` or `OmittableFieldBinding` would provide you with some more readable alternatives.